### PR TITLE
[DOC] Correct bounds for v in RectSphereBivariateSpline

### PR DESCRIPTION
**Pull request by sriki18**
This is a fix for #6556 . I changed the SciPy documentation to reflect the actual functionality of fftpack.

### CODE REVIEW

File: scipy/interpolate/fitpack2.py [hunk 1]
```diff
@@ -1529,7 +1529,7 @@ class RectSphereBivariateSpline(SphereBivariateSpline):
         (0, pi).
     v : array_like
         1-D array of longitude coordinates in strictly ascending order.
-        Coordinates must be given in radians, and must lie within (0, 2pi).
+        Coordinates must be given in radians, and must lie within (-pi, pi).
```
Comment (person142) [inline scipy/interpolate/fitpack2.py]: This isn't quite right--the FITPACK documentation requires that

- ``-pi <= v[0] < pi``
- ``v[i] < v[i+1]`` (the strictly ascending bit)
- ``v[-1] <= v[0] + 2*pi``.

(Changed the indexing from the documentation for python conventions.) So for example in #6556 the example
```python
>>> u = np.linspace(1, 3)
>>> v = np.linspace(np.pi - 1e-6, 6)
>>> r = np.zeros([u.size, v.size])
>>> RectSphereBivariateSpline(u, v, r)
```
works fine even though 6 is not in the interval ``(-pi, pi)``.

**Review by person142 — CHANGES_REQUESTED** [review]

**Comment by person142** [comment]
@sriki18 I wonder if you're planning to continue to working on this? No pressure, just don't want it to slip into the mists of time.

**Comment by sriki18** [comment]
Hi, Sorry I didn't catch your earlier comment. I am going to have a busy week and will have a go at it if I find the time. I should definitely find time by Friday. If you would prefer to work on it in the mean time, please let me know. Thanks!

**Comment by person142** [comment]
Sounds great. No rush, just wanted to make sure you were still interested.

**Comment by sriki18** [comment]
Working on this at the moment.

**Comment by pv** [comment]
superceded by #6665

**Pull request closed without merging**